library/socket/inet_pton.c

```diff
@@ -37,7 +37,8 @@ inet_pton(int af, const char *src, void *dst) {
 static int
 inet_pton4(const char *src, unsigned char *dst) {
     int saw_digit, octets, ch;
-    unsigned char tmp[NS_INADDRSZ], *tp;
+    unsigned char tmp[NS_INADDRSZ] = {0};
+    unsigned char *tp;
```

Comment:
The `tmp` was not initialised and later, based on conditions, an initial value is required